subsys/net/lib/http/http_server_core.c

```diff
@@ -704,6 +705,42 @@ struct http_resource_detail *get_resource_detail(const char *path,
 	return NULL;
 }
 
+int http_server_find_file(char *fname, size_t fname_len, size_t *file_size, bool *gzipped)
+{
+	struct fs_dirent dirent;
+	size_t len;
+	int ret;
+
+	ret = fs_stat(fname, &dirent);
+	if (ret < 0) {
+		len = strlen(fname);
+		snprintk(fname + len, fname_len - len, ".gz");
+		ret = fs_stat(fname, &dirent);
+		*gzipped = (ret == 0);
+	}
+
+	if (ret == 0) {
+		*file_size = dirent.size;
+		return ret;
+	}
+
+	return -ENOENT;
+}
+
+void http_server_get_content_type_from_extension(char *url, size_t url_len, char *content_type,
+						 size_t content_type_size)
+{
+	HTTP_SERVER_CONTENT_TYPE_FOREACH(ct)
+	{
+		char *ext = &url[url_len - ct->extension_len];
+
+		if (0 == strncmp(ext, ct->extension, ct->extension_len)) {
```

Comment:
Please don't use this style, we always do the check other way around, so `strncmp() == 0`